services/recipeSearchController.js

```diff
@@ -7,6 +7,336 @@ var Supermarket = require('../components/supermarket/supermarketSchema');
+                        }
+        //calculating the coverage for each recipe from the queryResult
+                        for (var recipeCounter = 0; recipeCounter < queryResult.length; recipeCounter++) {
+                            queryResult[recipeCounter].searchResult = compareLists(queryResult[recipeCounter].ingredientList, req.body.ingredients);
+                        }
+
+        //sort function for the recipes to sort them by their match ingredients (from high to low)
+                        if (queryResult.length > 1){
+                            queryResult.sort(function (a, b) {
+                                if (a.searchResult.match < b.searchResult.match) {
+                                    return 1;
+                                }
+                                if (a.searchResult.match > b.searchResult.match) {
+                                    return -1;
+                                }
+                                // a must be equal to b
+                                return 0;
+                            });
+                        }
+
+                        res.json(filteredResults);
+                    });
+
+                } else {
+
+
+                    //calculating the coverage for each recipe from the queryResult
+                    for (var recipeCounter = 0; recipeCounter < queryResult.length; recipeCounter++) {
+                        queryResult[recipeCounter].searchResult = compareLists(queryResult[recipeCounter].ingredientList, req.body.ingredients);
+                    }
+
+                    //sort function for the recipes to sort them by their match ingredients (from high to low)
+                    if (queryResult.length > 1){
+                        queryResult.sort(function (a, b) {
+                            if (a.searchResult.match < b.searchResult.match) {
+                                return 1;
+                            }
+                            if (a.searchResult.match > b.searchResult.match) {
+                                return -1;
+                            }
+                            // a must be equal to b
+                            return 0;
+                        });
+                    }
+
+                    res.json(queryResult);
+
+                }
+            });
+    });
+
+
+};
+
+//returns the % of the coverage of ingredients from list A by list B
+// (how many % of the ingredients of list A are in list B)
+
+function compareLists(ingredientListA, searchIngredientList) {
+    var matches = 0; //count of ingredients from the search that are in the given recipe
+    var notUsedIngredientcounter = 0; // counter to store the not used Ingredient in the right space
+    var notUsedIngredients = [];// array with the not used Ingredients from the search term
+
+    for (var counterB = 0; counterB < searchIngredientList.length; counterB++) { //for each ingredient from the search
+        var noMatch = 0;
```

Comment:
Irreführender Name: steht das für "number of Matches" oder für "no (kein) Match"